pkg/controllers/vrpq/query_reconciler.go

```diff
@@ -71,66 +80,106 @@ func (q *QueryReconciler) Reconcile(ctx context.Context, req *ctrl.Request) (ctr
-		)
-
-		return nil
-	})
+	}
 
 	return res, err
 }
 
-// runListRestorePoints will update the status condition before and after calling
-// list restore points api
-// Temporarily, runListRestorePoints will not call the ListRestorePoints API
-// since the dispatcher is not set up yet
-func (q *QueryReconciler) runListRestorePoints(ctx context.Context, _ *ctrl.Request) error {
+// runShowRestorePoints call the vclusterOps API to get the restore points
+func (q *QueryReconciler) runShowRestorePoints(ctx context.Context, dispatcher vadmin.Dispatcher,
+	opts []restorepointsquery.Option) (err error) {
 	// set Querying status condition and state prior to calling vclusterops API
-	err := vrpqstatus.UpdateConditionAndState(ctx, q.VRec.Client, q.VRec.Log, q.Vrpq,
+	err = vrpqstatus.UpdateConditionAndState(ctx, q.VRec.Client, q.VRec.Log, q.Vrpq,
 		v1.MakeCondition(vapi.Querying, metav1.ConditionTrue, "Started"), stateQuerying)
 	if err != nil {
 		return err
 	}
 
-	// API should be called to proceed here
-	// If we receive a failure result from the API, a state message and condition need to be updated
-
-	// clear Querying status condition
-	err = vrpqstatus.UpdateConditionAndState(ctx, q.VRec.Client, q.VRec.Log, q.Vrpq,
-		v1.MakeCondition(vapi.Querying, metav1.ConditionFalse, "Completed"), stateQuerying)
-	if err != nil {
-		return err
+	// call showRestorePoints vcluster API
+	q.VRec.Eventf(q.Vrpq, corev1.EventTypeNormal, events.ShowRestorePointsStarted,
+		"Starting show restore points")
+	start := time.Now()
+	if res, errRun := dispatcher.ShowRestorePoints(ctx, opts...); verrors.IsReconcileAborted(res, errRun) {
+		if errRun != nil {
+			err = vrpqstatus.UpdateConditionAndState(ctx, q.VRec.Client, q.VRec.Log, q.Vrpq,
+				v1.MakeCondition(vapi.Querying, metav1.ConditionFalse, "Failed"), stateFailedQuery)
+			if err != nil {
+				errRun = errors.Join(errRun, err)
+			}
+			return errRun
+		}
 	}
+	q.VRec.Eventf(q.Vrpq, corev1.EventTypeNormal, events.ShowRestorePointsSucceeded,
+		"Successfully queried restore points in %s", time.Since(start).Truncate(time.Second))
 
 	// set the QueryComplete if the vclusterops API succeeded
 	return vrpqstatus.UpdateConditionAndState(ctx, q.VRec.Client, q.VRec.Log, q.Vrpq,
 		v1.MakeCondition(vapi.QueryComplete, metav1.ConditionTrue, "Completed"), stateSuccessQuery)
 }
+
+// findRunningPodWithNMAContainer finds a pod to execute the vclusterops API.
+// The pod should be running and the NMA container should be ready
+func (q *QueryReconciler) findRunningPodWithNMAContainer(pods *corev1.PodList) (string, error) {
+	for i := range pods.Items {
+		pod := &pods.Items[i]
+		if pod.Status.Phase == podRunning {
+			for j := range pod.Status.ContainerStatuses {
+				if pod.Status.ContainerStatuses[j].Ready && pod.Status.ContainerStatuses[j].Name == names.NMAContainer {
+					return pod.Status.PodIP, nil
+				}
+			}
+		}
+	}
+	return "", fmt.Errorf("can't find a pod to call the vclusterOps API")
```

Comment:
We shouldn't surface this as an error. We should write a message to the log saying we couldn't find any pod to run the query at and requeue the iteration.